File: src/code/python/sorting_algorithms/insertion_sort.py
def insertion_sort(arr: list) -> None:
    n = len(arr)

    for i in range(1, n):
        key = arr[i]

        while i > 0 and key < arr[i - 1]:
            arr[i] = arr[i - 1]
            i -= 1

        arr[i] = key

File: src/code/ruby/sorting_algorithms/insertion_sort.rb
def insertion_sort(arr)
  n = arr.size

  (1...n).each do |i|
    key = arr[i]
    j = i - 1

    while j >= 0 && key < arr[j]
      arr[j + 1] = arr[j]
      j -= 1
    end

    arr[j + 1] = key
  end
end
